### DISCUSSION TITLE
Display Ideas for Relationships - Minimum, Maximum, Index

### DISCUSSION THREAD

**Discussion by StephenGunn** [directus/directus#9445] [Feature Requests]
Just want to start this out by saying I love Directus. It's very exciting to find such a great piece of software being actively updated.

I have an app that I am converting from WordPress that is an art challenge community website. It has some specific needs that makes me really long for a few features that either don't exist or I can't figure out how to implement in the admin. 

I'll start by explaining my situation and why I might want these kinds of features. I might be missing how to set them up with the current system. 

I have a bunch of 'parameters', which are in 4 individual models. I also have a singleton with 4 one-to-many relationships to manage which parameters are active. There will always be 24 total parameters in each relationship. The order and number matter. 

It would be a huge help for my non-techie admin to be able to see how many items are in each relationship with out having to count. Having a max amount and a min amount would be very nice. Having the index displayed on the relationship item would be very nice. Disabling the Add button would also work.

I have the sort field in the item display which kind of works, but it has to be set. Index would be automatic... I think?

Here is what I have so far (the parameters are not added in yet since I am still setting up the models)
![image](https://user-images.githubusercontent.com/7240548/140207152-d2ab531d-7864-4795-b170-bad464f854c9.png)

Maybe I am missing how to set this up or maybe it can't be. Either way, love the software.

**Comment by rijkvanzanten** [comment]
So if I'm understanding correctly, you'd like to see a little "3 items selected" message at the top of each of those fields right? That's not currently a thing, but could be a useful addition!

**Reply by StephenGunn** [reply]
That would be a great fix for seeing how many items are in the relationship.

Being able to disable adding new items once there are N added would also be ideal for my case.

Thank you.

**Reply by rijkvanzanten** [reply]
The second should be fairly doable to implement, without requiring more in-depth discussions on the ideal UX/UI 🙂 Wanna try making a PR for that?

**Reply by StephenGunn** [reply]
I can try, I haven't used Vue in about a year but I would love to contribute.

It would be setting N in the Conditions, right?

**Reply by rijkvanzanten** [reply]
I think we'd want to do that through an interface option for the relational interfaces first (eg in https://github.com/directus/directus/blob/main/app/src/interfaces/list-o2m/index.ts). That allows the user to set the max number, after which we can simply disable the "select existing" and "add new" buttons. It's not security of course, but a useful UX addition. Next to that, and this is more future-material as it's way more involved, we should discuss how we can set that same thing up as a permission in the API, to really lock it down from a security perspective as well 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
